src/Psalm/Internal/ClauseConjunction.php

```diff
@@ -0,0 +1,730 @@
+<?php
+
+namespace Psalm\Internal;
+
+use Psalm\Exception\ComplicatedExpressionException;
+use Psalm\Storage\Assertion;
+use Psalm\Storage\Assertion\Falsy;
+
+use function array_filter;
+use function array_intersect_key;
+use function array_keys;
+use function array_merge;
+use function array_pop;
+use function array_values;
+use function count;
+use function implode;
+use function in_array;
+use function mt_rand;
+use function reset;
+
+/**
+ * Represents a CNF expression ((A||B) && (C||D) && ...)
+ *
+ * @psalm-immutable
+ *
+ * @internal
+ */
+final class ClauseConjunction
+{
+    /**
+     * @var list<Clause>
+     */
+    public array $clauses;
+
+    /**
+     * Negation of the current clause.
+     */
+    private ?self $negation = null;
+
+    public bool $isSimplified = false;
+
+    private static ?self $empty = null;
```

Comment:
```suggestion
    private ?ClauseConjunction $negation = null;

    public bool $isSimplified = false;

    private static ?ClauseConjunction $empty = null;
```

I have a hunch the above may resolve phar build issue

Reply:
huh, let's give it a try!

Reply:
And it seems it did.

Reply:
The master now has `humbug/box` updated, which does not exhibit this issue.